examples/shouldi/shouldi/dependency_check.py

```diff
@@ -0,0 +1,47 @@
+import json
+import asyncio
+from typing import Dict, Any
+
+from dffml.df.base import op
+from dffml.df.types import Definition
+
+package_src_dir = Definition(name="package_src_dir", primitive="str")
+dependency_check_output = Definition(
+    name="dependency_check_output", primitive="Dict[str, Any]"
+)
+
+
+@op(
+    inputs={"pkg": package_src_dir},
+    outputs={"report": dependency_check_output},
+)
+async def run_dependency_check(pkg: str) -> Dict[str, Any]:
+    """
+    CLI usage: dffml service dev run -log debug shouldi.dependency_check:run_dependency_check -pkg .
+    """
+    proc = await asyncio.create_subprocess_exec(
+        "dependency-check.sh",
+        "-f",
+        "JSON",
+        "-s",
+        ".",
+        cwd=pkg,
+        stdout=asyncio.subprocess.PIPE,
+        stderr=asyncio.subprocess.PIPE,
+    )
+
+    with open(pkg + "/dependency-check-report.json") as f:
```

Comment:
Lets create a temporary directory `with tempdir.TemporaryDirectory` and use it with the `--out` flag to keep the output in another directory. Since we can't know if the directory we'll be pointed at is writable or not, whereas a tempdir will always be writeable.

![image](https://user-images.githubusercontent.com/5950433/79365175-1c03e900-7eff-11ea-81c1-27a26a9b3509.png)